utils/bazel/llvm-project-overlay/lldb/BUILD.bazel

```diff
@@ -728,37 +728,57 @@ cc_library(
     ],
 )
 
-cc_library(
-    name = "liblldb.static",
-    deps = [
-        ":API",
-        ":Interpreter",
-    ],
+genrule(
+    name = "gen_exports_file_linux",
+    srcs = ["//lldb:source/API/liblldb-private.exports"],
+    outs = ["exports_linux.txt"],
+    cmd = """
+cat > $(OUTS) <<EOF
+{
+  global:
+    $$(sed 's/$$/;/g' $(SRCS))
+};
+EOF
+""",
 )
 
-cc_shared_library(
-    name = "liblldb",
-    # TODO: Remove once fixed https://github.com/bazelbuild/bazel/issues/21893
+genrule(
+    name = "gen_exports_file_macos",
+    srcs = ["//lldb:source/API/liblldb-private.exports"],
+    outs = ["exports_macos.txt"],
+    cmd = "sed 's/^/_/g' $(SRCS) > $(OUTS)",
+)
+
+cc_binary(
+    name = "lldb{}".format(PACKAGE_VERSION),
```

Comment:
Not sure whether `select` works in the `name` attribute. If it does, how about naming this `"liblldb{}.so".format(PACKAGE_VERSION)` (or `.dylib`) explicitly?

On another note, the fact that we're using a `cc_binary` here is a fairly subtle detail. It might make sense to have a short comment explaining why we can't use `cc_shared_library` or `cc_library`.

Reply:
select doesn't work on name, but also this actually already does output the correct name for the library. with cc_binary + linkshared you  don't have control over the output file name and it implicitly adds `lib` and `.dylib`/`.so`, so actually we're lucky here that we can name it with this version in the name so it doesn't conflict with the `lldb` target. we could write our own trivial rule on top of the C++ bazel APIs to make this nicer, but probably not worth it. will add a comment